components/sftp/watch-remote-dir-for-new-file.js

```diff
@@ -0,0 +1,33 @@
+const sftpPipedreamApp = require('https://github.com/skilbjo/pipedream/components/sftp/sftp.app.js'); // TODO s/skilbjo/PipedreamHQ/
+
+module.exports = {
+  name: "watch a remote directory for a new file",
+  version: "0.0.1",
+  props: {
+    db: "$.service.db",
+    sftpPipedreamApp,
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 5,
+      },
+    }
+  },
+  async run(event) {
+    const sftp = await src.methods.getClient({
```

Comment:
I'd recommend removing the explicit passing of `this.$auth` here and rely on the logic you have in `getClient`. I believe even with the code you have now, it should fall back to using the user's linked account (via `this.$auth`), and the small refactor I suggested should work in the same way.